set a default value for requested capacity for consumer deployment

leelavg opened the pull request:
Signed-off-by: Leela Venkaiah G <lgangava@redhat.com>

leelavg commented:
```
# make test                                                                                                                                                                                  
/root/go/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
api/v1alpha1/zz_generated.deepcopy.go
go vet ./...
/root/go/bin/controller-gen "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
mkdir -p /root/toolbox/orc/ocs-osd-deployer/testbin
test -f /root/toolbox/orc/ocs-osd-deployer/testbin/setup-envtest.sh || curl -sSLo /root/toolbox/orc/ocs-osd-deployer/testbin/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.6.3/hack/setup-envtest.sh
source /root/toolbox/orc/ocs-osd-deployer/testbin/setup-envtest.sh; fetch_envtest_tools /root/toolbox/orc/ocs-osd-deployer/testbin; setup_envtest_env /root/toolbox/orc/ocs-osd-deployer/testbin; go test ./... -coverprofile cover.out
Using cached envtest tools from /root/toolbox/orc/ocs-osd-deployer/testbin
setting up env vars
?       github.com/red-hat-storage/ocs-osd-deployer     [no test files]
?       github.com/red-hat-storage/ocs-osd-deployer/api/v1alpha1        [no test files]
ok      github.com/red-hat-storage/ocs-osd-deployer/controllers 247.035s        coverage: 84.8% of statements
?       github.com/red-hat-storage/ocs-osd-deployer/readinessProbe      [no test files]
ok      github.com/red-hat-storage/ocs-osd-deployer/readinessProbe/readiness    16.661s coverage: 73.3% of statements
?       github.com/red-hat-storage/ocs-osd-deployer/templates   [no test files]
?       github.com/red-hat-storage/ocs-osd-deployer/testutils   [no test files]
?       github.com/red-hat-storage/ocs-osd-deployer/utils       [no test files]
```

nb-ohad commented on controllers/managedocs_controller.go:
```diff
@@ -817,8 +817,12 @@ func (r *ManagedOCSReconciler) getDesiredProviderStorageCluster() (*ocsv1.Storag
+
+	// TODO: set max possible quota size
+	// re-visit if product enforces quota or any means respecting the storage claim
+	storageSize := fmt.Sprintf("%v", int32(math.MaxInt32))
+	storageUnit := "Ti"
```
First, I think this is the wrong place to do the overwrite, I would prefer to override the addon param when we read it and not the local var we are using for calculations, mainly because we want to see/track the actual value we use in the various log message we print with addon params values.

Second, please limit the storage size to 1 or 2 Peta, any value bigger than that might cause issues in rook or ceph as they usually add a little bit on top (which is hard to do when you give the max value of the int type)

Taking both of the above points into consideration, I would suggest to implement the following change here https://github.com/red-hat-storage/ocs-osd-deployer/blob/main/controllers/managedocs_controller.go#L512

```
		for key, value := range addonParamSecret.Data {
			r.addonParams[key] = string(value)
		}
		
		// Deprecting consumer requested capacity addon parameter by overriding it 
		// with a very high pre-defined value
		if r.DeploymentType == consumerDeploymentType {
			r.Log.Info(fmt.Printf(
				`Overriding addon parameter "%s" (currently "%s") with 1024`,
				storageSizeKey,
				r.addonParams[storageSizeKey] 
			))
			r.addonParams[storageSizeKey] = "1024"

			r.Log.Info(fmt.Printf(
				`Overriding addon parameter "%s" (currently "%s") with Ti`,
				storageUnitKey,
				r.addonParams[storageUnitKey] 
			))
			r.addonParams[storageUnitKey] = "Ti"
		}
```

leelavg replied:
Looks neat and thanks for explanations.

nb-ohad commented on controllers/managedocs_controller_test.go:
```diff
@@ -618,9 +628,8 @@ var _ = Describe("ManagedOCS controller", func() {
-						Skip(fmt.Sprintf("Skipping the test as it is not required by %v", testReconciler.DeploymentType))
-					}
+					// quota size isn't enforced by provider when consumer supplies it
+					Skip(fmt.Sprintf("Skipping the test as it is not required by %v", testReconciler.DeploymentType))
```
Why keep the test if we deprecated the `unit` addon parameter?

leelavg replied:
Ack, I'll remove standalone consumer tests involving size & unit.

nb-ohad commented on controllers/managedocs_controller_test.go:
```diff
@@ -732,8 +741,7 @@ var _ = Describe("ManagedOCS controller", func() {
 					case consumerDeploymentType:
-						secret.Data["size"] = []byte("1")
-						secret.Data["unit"] = []byte("Ti")
+						// "size" and "unit" params are temporarily not needed for consumer deployment
```
```suggestion
						// "size" and "unit" params are deprecated for consumer deployment
```

leelavg commented on api/v1alpha1/zz_generated.deepcopy.go:
```diff
@@ -1,3 +1,4 @@
+//go:build !ignore_autogenerated
```
- auto generated when `make generate` was run with `go1.16.15 linux/amd64`

nb-ohad replied:
ok

leelavg commented:
- addressed comments and tests are re-run
```
# make test                                                                                                                                                                                    
/root/go/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."                                                                                                          
go fmt ./...                                                                                                                                                                                   
go vet ./...                                                                                                                                                                                   
/root/go/bin/controller-gen "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases                                           
mkdir -p /root/toolbox/orc/ocs-osd-deployer/testbin                                                                                                                                            
test -f /root/toolbox/orc/ocs-osd-deployer/testbin/setup-envtest.sh || curl -sSLo /root/toolbox/orc/ocs-osd-deployer/testbin/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs
/controller-runtime/v0.6.3/hack/setup-envtest.sh                                                                                                                                               
source /root/toolbox/orc/ocs-osd-deployer/testbin/setup-envtest.sh; fetch_envtest_tools /root/toolbox/orc/ocs-osd-deployer/testbin; setup_envtest_env /root/toolbox/orc/ocs-osd-deployer/testbi
n; go test ./... -coverprofile cover.out                                                                                                                                                       
Using cached envtest tools from /root/toolbox/orc/ocs-osd-deployer/testbin                                                                                                                     
setting up env vars                                                                                                                                                                            
?       github.com/red-hat-storage/ocs-osd-deployer     [no test files]                                                                                                                        
?       github.com/red-hat-storage/ocs-osd-deployer/api/v1alpha1        [no test files]                                                                                                        
ok      github.com/red-hat-storage/ocs-osd-deployer/controllers 246.561s        coverage: 84.8% of statements                                                                                  
?       github.com/red-hat-storage/ocs-osd-deployer/readinessProbe      [no test files]
ok      github.com/red-hat-storage/ocs-osd-deployer/readinessProbe/readiness    16.263s coverage: 73.3% of statements
?       github.com/red-hat-storage/ocs-osd-deployer/templates   [no test files]
?       github.com/red-hat-storage/ocs-osd-deployer/testutils   [no test files]
?       github.com/red-hat-storage/ocs-osd-deployer/utils       [no test files]
```
